SqlDatabaseChain with Postgres implementation

khoroshevj opened the pull request:
https://github.com/tryAGI/LangChain/issues/73
https://github.com/tryAGI/LangChain/issues/75

there is no [SQLAlchemy](https://www.sqlalchemy.org/) like lib for .net, so I added SqlDatabase base class and for each db type separate implementation needed.

added LangChain.Utilities.PostgresDatabases.IntegrationTests with SqlDatabaseChain test using Postgres+OpenAI

HavenDV commented:
Very cool code quality.
Maybe we can simplify the project names a little (as a result, the names of NuGet packages) - instead of LangChain.Utilities.PostgresDatabases just LangChain.Utilities.Postgres and the same for SqlDatabases?

HavenDV commented on src/libs/Utilities/LangChain.Utilities.SqlDatabases/LangChain.Utilities.SqlDatabases.csproj:
```diff
@@ -0,0 +1,21 @@
+
+    <PropertyGroup Label="NuGet">
+        <Description>LangChain Utilities SqlDatabase.</Description>
+        <PackageTags>$(PackageTags);utilities;sqldatabase</PackageTags>
```
utilities PackageTag already specified in Directory.Build.props

HavenDV commented on src/libs/Utilities/LangChain.Utilities.PostgresDatabases/LangChain.Utilities.PostgresDatabases.csproj:
```diff
@@ -0,0 +1,25 @@
+
+    <PropertyGroup Label="NuGet">
+        <Description>LangChain Utilities PostgresDatabase.</Description>
+        <PackageTags>$(PackageTags);utilities;sqldatabase;postgres</PackageTags>
```
utilities PackageTag already specified in Directory.Build.props

khoroshevj commented:
changed project names and added few more tests.
btw, it is not fully implemented, I'll add a comment to https://github.com/tryAGI/LangChain/issues/75 so we can merge this and implement rest later

khoroshevj commented:
@HavenDV comments resolved, everything else fine?
